Client/Frontend/Browser/OpenPdfHelper.swift

```diff
@@ -0,0 +1,150 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import WebKit
+import SnapKit
+
+import Shared
+
+import XCGLogger
+
+private let log = Logger.browserLogger
+
+struct OpenInViewUX {
+    static let ViewHeight: CGFloat = 40.0
+    static let TextFont = UIFont.systemFontOfSize(16)
+    static let TextColor = UIColor(red: 74.0/255.0, green: 144.0/255.0, blue: 226.0/255.0, alpha: 1.0)
+    static let TextOffset = -15
+}
+
+enum FileType : String {
+    case PDF = "pdf"
+}
+
+protocol OpenInHelper {
+    static func canOpen(url: NSURL) -> Bool
+    func showOpenInView(inView parentView: UIView, forWebView webView: WKWebView)
+    func hideOpenInView(forWebView webView: WKWebView)
+    func open()
+}
+
+struct OpenInHelperFactory {
+    static func helperForURL(url: NSURL) -> OpenInHelper? {
+        if OpenPdfInHelper.canOpen(url) {
+            return OpenPdfInHelper(url: url)
+        }
+
+        return nil
+    }
+}
+
+class OpenPdfInHelper: NSObject, OpenInHelper, UIDocumentInteractionControllerDelegate {
+    private var view: OpenInView?
+    private var docController: UIDocumentInteractionController? = nil
+
+    private var parentView: UIView!
+    private var webView: WKWebView!
+
+    init(url: NSURL) {
+        super.init()
+        let contentsOfFile = NSData(contentsOfURL: url)
+        let dirPaths = NSSearchPathForDirectoriesInDomains(.DocumentDirectory,
+            .UserDomainMask, true)
+
+        let dirPath = NSURL(fileURLWithPath: dirPaths[0]).URLByAppendingPathComponent("pdfs")
+        let fileManager = NSFileManager.defaultManager()
+        do {
+            guard let lastPathComponent = url.lastPathComponent else {
+                log.error("failed to create proper URL")
+                return
+            }
+            let filePath = dirPath.URLByAppendingPathComponent(lastPathComponent)
+            try fileManager.createDirectoryAtURL(dirPath, withIntermediateDirectories: true, attributes: nil)
+            if fileManager.createFileAtPath(filePath.absoluteString, contents: contentsOfFile, attributes: nil) {
```

Comment:
What happens to this file after we create it? Does it get deleted anywhere? Would it make more sense to use `/tmp`?